kpops/components/streams_bootstrap/streams/streams_app.py

```diff
@@ -24,6 +30,26 @@ class StreamsAppCleaner(KafkaAppCleaner):
     def helm_chart(self) -> str:
         return f"{self.repo_config.repository_name}/{AppType.CLEANUP_STREAMS_APP.value}"
 
+    @cached_property
+    def pvc_handler(self) -> PVCHandler:
+        return PVCHandler(self.full_name, self.namespace)
+
+    @override
+    async def clean(self, dry_run: bool) -> None:
+        await super().clean(dry_run)
+        if self.app.stateful_set and self.app.persistence.enabled:
+            await self.clean_pvcs(dry_run)
+
+    async def clean_pvcs(self, dry_run: bool) -> None:
+        if dry_run:
+            pvc_names = await self.pvc_handler.get_pvc_names()
+            log.info(
+                f"Deleting the PVCs {pvc_names} for StatefulSet '{self.full_name}'"
+            )
```

Comment:
how about moving the dry_run case to the PVC handler as well?

Reply:
I would disagree. I would keep the functionality of the PVCHandler limited to Kubernetes and night tight to KPOps dry_run.